CodeEditor/Sources/CodeEditor/UXCodeTextView.swift

```diff
@@ -338,6 +336,84 @@ final class UXCodeTextView: UXTextView, HighlightDelegate, UIScrollViewDelegate
     }
+    
+    func updateLightBulbs() {
+        // delete and add new lightbulbs when code changes
+        for lightBulb in lightBulbs {
+            lightBulb.removeFromSuperview()
+        }
+        lightBulbs = []
+        addLightbulbs()
+    }
+    
+    // calculates the offset of the given line if line wrapping occurs
+    private func calculateWrapOffsetOf(_ lineNumber: Int) -> Int {
+        var offset = 0, curr = 1, subParaCount = 0
+        var subParaRange = NSRange(location: 0, length: 0)
+        layoutManager.enumerateLineFragments(forGlyphRange: layoutManager.glyphRange(for: textContainer)) { _, _, _, glyphRange, stop in
+            if curr == lineNumber {
+                let charRange = self.layoutManager.characterRange(forGlyphRange: glyphRange, actualGlyphRange: nil)
+                let paraRange = NSString(string: self.textStorage.string).paragraphRange(for: charRange)
+                if subParaRange != paraRange {
+                    // if done during subPara jump back to actual line
+                    offset -= subParaCount
+                }
+                stop.pointee = true
+            } else {
+                let charRange = self.layoutManager.characterRange(forGlyphRange: glyphRange, actualGlyphRange: nil)
+                if subParaRange.length > 0 {
+                    subParaRange.length += charRange.length
+                } else {
+                    subParaRange = charRange
+                }
+                let paraRange = NSString(string: self.textStorage.string).paragraphRange(for: charRange)
+                if subParaRange != paraRange {
+                    offset += 1
+                    subParaCount += 1
+                } else {
+                    subParaRange = NSRange(location: 0, length: 0)
+                    subParaCount = 0
+                }
+                curr += 1
+            }
+        }
+        return offset
+    }
+    
+    func addLightbulbs() {
+        var lineNumber = 1
+        layoutManager.enumerateLineFragments(forGlyphRange: layoutManager.glyphRange(for: textContainer)) { rect, _, _, _, _ in
+            let offset = self.calculateWrapOffsetOf(lineNumber)
+            if let feedback = self.feedbackSuggestions.first(where: { $0.fromLine == lineNumber - offset }) {
+                self.currFeedbackId = feedback.id.uuidString
+                let button = UIButton()
+                let image = UIImage(systemName: "lightbulb.fill")
+                button.setImage(image, for: .normal)
+                button.imageView?.contentMode = .scaleAspectFit
+                button.imageView?.tintColor = .yellow
+                button.frame = CGRect(
+                    x: 5,
+                    y: rect.origin.y,
+                    width: 25,
+                    height: 25
+                )
+                .offsetBy(dx: 0, dy: 8)
+                button.addTarget(self, action: #selector(self.onLightBulbTap), for: .touchUpInside)
```

Comment:
Can we have a helper function to create this button? Something like `createLightbulbButton`? Maybe even an own view or whatever that is in UIKit.